lms/djangoapps/courseware/views/views.py

```diff
@@ -1485,6 +1487,34 @@ def enclosing_sequence_for_gating_checks(block):
     return None
 
 
+def _check_sequence_exam_access(request, location):
+    """
+    Checks the client request for an exam access token for a sequence.
+    Exam access is always granted at the sequence block. This method of gating is
+    only used by the edx-exams system and NOT edx-proctoring.
+    """
+    if not exams_ida_enabled(location.course_key):
+        return True
+
+    if request.user.is_staff or is_masquerading_as_specific_student(request.user, location.course_key):
+        return True
+
+    exam_access_token = request.COOKIES.get(settings.EXAM_ACCESS_TOKEN_COOKIE)
+    if exam_access_token:
+        try:
+            exam_access_unpacked = unpack_token_for(exam_access_token, request.user.id)
+        except:  # pylint: disable=bare-except
+            log.exception("Failed to validate exam access token")
+            return False
+
+        return (
+            str(location) == exam_access_unpacked.get('content_id')
+            and request.user.id == exam_access_unpacked.get('lms_user_id')
```

Comment:
Isn't the check for the request.user.id being equal to the lms_user_id redundant here since unpack_token_for already validates this?
https://github.com/edx/token-utils/blob/main/token_utils/unpack.py#L32

Reply:
good catch, yeah this isn't needed